panoptes_cli/commands/subject.py

```diff
@@ -49,31 +52,64 @@ def ls(subject_set_id, quiet, subject_ids):
 
 
 @subject.command()
-@click.argument('subject-id', required=True)
+@click.argument("subject-id", required=True)
 def info(subject_id):
     subject = Subject.find(subject_id)
     click.echo(yaml.dump(subject.raw))
 
 
 @subject.command()
 @click.option(
-    '--force',
-    '-f',
+    "--force",
+    "-f",
     is_flag=True,
-    help='Delete without asking for confirmation.',
+    help="Delete without asking for confirmation.",
 )
-@click.argument('subject-ids', required=True, nargs=-1, type=int)
+@click.argument("subject-ids", required=True, nargs=-1, type=int)
 def delete(force, subject_ids):
     for subject_id in subject_ids:
         if not force:
-            click.confirm('Delete subject {}?'.format(subject_id), abort=True)
+            click.confirm("Delete subject {}?".format(subject_id), abort=True)
         Subject.find(subject_id).delete()
 
 
+@subject.command()
+@click.option(
+    "--replace",
+    "-r",
+    is_flag=True,
+    help="Replace all existing metadata rather than merging.",
+)
+@click.argument("metadata-file", required=True, nargs=1)
+def update_metadata(replace, metadata_file):
+    """
+    Updates subject metadata from a CSV file.
+
+    The CSV file should contain a "subject_id" column listing subject IDs.
+    All other column names are taken to be metadata keys.
+    """
+    with open(metadata_file, "r") as metadata_f:
+        total_tows = len(metadata_f.readlines())
+        metadata_f.seek(0)
+        metadata_rows = csv.DictReader(metadata_f)
+        with click.progressbar(
+            metadata_rows,
+            length=total_tows,
+            label="Update subject metadata",
+        ) as _metadata_rows:
+            for metadata in _metadata_rows:
+                subject_id = metadata.pop("subject_id")
+                try:
+                    subject = Subject.find(subject_id)
+                except Exception as e:
+                    click.echo(f"Failed to update subject {subject_id}: {e}", err=True)
+                if replace:
+                    subject.metadata = dict(metadata)
+                else:
+                    subject.metadata.update(metadata)
+                subject.save()
```

Comment:
```suggestion
                try:
                    subject = Subject.find(subject_id)
                    if replace:
                        subject.metadata = dict(metadata)
                    else:
                        subject.metadata.update(metadata)
                    subject.save()
                except Exception as e:
                    click.echo(f"Failed to update subject {subject_id}: {e}", err=True)
                
```

Reply:
My above suggestion aims to catch multiple failures: 
1. Can't find subject via `find()` (not likely)
2. Unsuccessful `save()` (most likely)
3. Any issues with replace or update of metadata (least likely)